### DISCUSSION TITLE
BlueSCSI and Atari ST ICD Hostadapter ...

### DISCUSSION THREAD

**Discussion by franklukas** [BlueSCSI/BlueSCSI-v2#167] [Help]
----------- bluescsi.ini ----
[SCSI]
System=Generic
EnableSCSI2=0 ; Off
EnableParity=1 ; On
Debug=1 ; On
--------------- or alternativ ---
[SCSI]
System=MegaSTE
MapLunsToIDs=1 ; On
EnableParity=0 ; Off
Debug=1 ; On

... with both ini files the BlueSCSI do not work with any ICD Atari ACSI/SCSI Hostadapter. Hard drives larger than 1GB, e.g. 9GB or 18GB hard drives, work on an ICD Host adapter. Parity is supported. 

On the same Host adapter a ZULU works perfectly and without problems. The ZULU works without any config file (ini) only with the HD image on the SD card.

If the Atari Computer has a real SCSI Controller like a Falcon or Atari TT there are no problems ...

You can not use any ID or LUN ...

![Hddriver](https://github.com/user-attachments/assets/adeaabcc-d52a-4877-9b03-823f1fb33533)

**Comment by mynameistroy** [comment]
Hey Frank,

I'm looking into this now with some hardware from BikerBob and hopefully I can figure out what is going on. There are some other strange issues with earlier era Sparcstations as well which might be caused by the same issue.

**Comment by mynameistroy** [comment]
It seems to only be specific ICD adapters oddly as I've tried the The Link II without issue and I have to retest but I believe my AdSCSI Pro adapter also worked.

**Reply by franklukas** [reply]
With my LINK97 Hostadapter it works fine with a Standard Atari MegaST with TOS 1.04.

![BlueSCSI_Link97](https://github.com/user-attachments/assets/a8e37833-f34a-4a1c-945b-ad4accf4118e)
. 


bluescsi.ini

[SCSI]
Debug=1
WiFiSSID="Fritz!Box-Frank"
WiFiPassword="31305831136958519762"
WiFiMACAddress=00:80:19:C0:FF:EE

**Reply by franklukas** [reply]
Unfortunately, the Bluescsi does not put Termpower on the SCSI bus. An ICD LINK or LINK2 or a LINK96/97 need Termpower on the bus to work.

According to the SCSI standard, several devices can put Termpower on the bus, decoupled by a Diode and protected by a 1 amp fuse.

**Reply by mynameistroy** [reply]
The v2 BlueSCSI does have a backfeed jumper to put power from the berg connector onto the bus to power devices like the LINK2.

**Reply by franklukas** [reply]
Yes "Back Feed" Jumper ...

**Reply by franklukas** [reply]
I always used a BlueSCSI 25 D-Sub Module on the Atari and the ICD Avantage Host adapter and it has the problems described above but the BlueSCSI Desktop version works perfectly and without problems ...

25 D-Sub Version ...
![25P_BlueSCSI](https://github.com/user-attachments/assets/6b1d8fe4-15eb-4a02-b9c3-72e426c0d35e)

Desktop Version ...
![Desktop_BlueSCSI](https://github.com/user-attachments/assets/92dab4f6-451c-46c5-bbdd-8ac95cb87c2c)

... used the same SD card with both BlueSCSI Versions.

**Reply by franklukas** [reply]
On a Atari TT with a real SCSI port the 25 D-Sub BlueSCSI works fine ...

**Reply by mynameistroy** [reply]
Interesting! I don't have a DB-25 model yet but one is on it's way so I can test with it

**Comment by mynameistroy** [comment]
So I started my barrage of testing and interestingly enough I found that BlueSCSI v2, ZuluSCSI RP2040 and ZuluSCSI 1.1+ all fail in the same way using the Tori MegaSTE adapter as built by BikerBob. The BlueSCSI XCVR v1 works perfectly without issue though.

It's strange because in all the "failing" boards it would seem they all read the driver and load it just fine but when it gets to the TOS desktop then clicking on the C drive results in may be corrupt/damaged error. I also see no SCSI traffic at all on the lines so it's like the adapter isn't even doing anything when asked.

I have external ICD clones to test as well plus 2 AdSCSI cards (earlier and later revs), plus the Link 2 so I will continue to post my findings.

This is all done currently with HDDRIVER 12 but I will likely do the same tests with ICD drivers as well

**Comment by franklukas** [comment]
There are some problems when operating the BlueSCSI v2 on the Atari TOS computer.

On the Falcon I use a 25 D-Sub BlueSCSI v2 and with MiNT and the SCSILINK.XIF Driver, the WLAN only works without Dayna entries in the ini and with the help of the new BlueSCSI Accessory from Thorsten Otto -> https://github.com/th-otto/wifi_da/actions/runs/10098442601

On my MegaSTE ICD Avantage Hostadapter from tOri the BlueSCSI v2 Desktop works fine also with a LINK96/97 Hostadapter but the 25 D-Sub v2 BlueSCSI has problems.

I use HDDriver 12 too ...

**Reply by mynameistroy** [reply]
I unfortunately do not have an external BlueSCSI v2 to test any of this out yet. Hopefully soon I will have one to verify some of the problems. Not sure why the Dayna entries in the ini work different then with Thorsten's tool though. It essentially does the same work under the hood.

Thanks for all the information though! Hopefully it can be figured out and fixed.

**Reply by erichelgeson** [reply]
@franklukas didn't we solve this ssid/pw issue a few weeks ago? https://github.com/BlueSCSI/BlueSCSI-v2/discussions/165#discussioncomment-10086821 - was there another issue? If so can you please re-run the debug firmware to validate as you did before.

**Reply by franklukas** [reply]
@erichelgeson ... yes the problem is solve with the ssid/pw issue with the use on a Macintosh IIsi.

But there are problems with Wi-Fi and the Atari TOS Computer. STING with the daynaport driver do not work at single TOS. Under MiNT with the SCSILINK.XIF works with the Atari Accessory from Thorsten Otto but do not work over the bluescsi.ini ...


@mynameistroy ... I will wait and see what comes up.

**Reply by erichelgeson** [reply]
@franklukas can you re-run that firmware in the linked thread above? I'm sure the issue will be apparent - again the ini and the bluescsi toolbox api do the same thing so it is /likely/ data entry - which the debug firmware I built will show. If not we can investigate.

**Reply by franklukas** [reply]
... it is about the 25 pin D-Sub v2 version use the SCSI drive section from the BlueSCSI with Atari SCSI ICD Hostadapter!

The Wi-Fi fault is a problem with the Atari Daynaport Driver because when I use the original Dayna SCSILINK Hardware all is fine with the Atari use single TOS with STING or MiNT.


Please describe exactly what I should do with which BlueSCSI adapter and which computer and which operating system?

**Reply by erichelgeson** [reply]
> Under MiNT with the SCSILINK.XIF works with the Atari Accessory from Thorsten Otto but do not work over the bluescsi.ini ...

If I'm reading it correctly that seems to be the same issue on the Atari as you had in the other thread on your mac? If so please just follow the same steps you did in the other thread with the Atari.

**Reply by franklukas** [reply]
no !!!

The Atari with MiNT and the SCSILINK.XIF driver freeze when I use the bluescsi.ini and the Wi-Fi entries in it. The SCSILINK.XIF works well with the original Dayna Hardware. Without the Wi-Fi entries in the ini and the use from the Wi-Fi DA from Thorsten Otto works.

There seems to be an important difference between the Macintosh driver and the SCSILINK.XIF for Atari MiNT and the STING Single TOS driver ...?

**Reply by erichelgeson** [reply]
Sorry maybe something is being lost in translation but isn't this exactly the issue you had on the mac?
> Without the Wi-Fi entries in the ini and the use from the Wi-Fi DA from Thorsten Otto works.

**Comment by franklukas** [comment]
@erichelgeson 

Read ...
https://github.com/freemint/freemint/issues/342
